pkg/kubelet/rkt/rkt.go

```diff
@@ -789,7 +789,12 @@ func (r *runtime) PullImage(image kubecontainer.ImageSpec, pullSecrets []api.Sec
 		tag = "latest"
 	}
 
-	creds, ok := r.dockerKeyring.Lookup(repoToPull)
+	keyring, err := credentialprovider.MakeDockerKeyring(pullSecrets, r.dockerKeyring)
```

Comment:
> Can you add a comment explaining why a docker keyring appears in a rkt file?

To be honest, I have no idea.  I updated the code to make use of the extended keyring, but I did not attempt to lookup to see why the rkt code used them to begin with.  If it was right before, it's still right.  If it was wrong before, it's no more wrong than it was.